src/lib/expansion-panel/expansion-summary.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  Component,
+  Input,
+  ViewEncapsulation,
+} from '@angular/core';
+import {
+  trigger,
+  state,
+  style,
+  transition,
+  animate,
+} from '@angular/animations';
+import {MdExpansionPanel, EXPANSION_PANEL_ANIMATION_TIMING} from './expansion-panel';
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-summary, mat-summary',
+  styleUrls: ['./expansion-summary.css'],
+  templateUrl: './expansion-summary.html',
+  encapsulation: ViewEncapsulation.None,
+  host: {
+    'tabindex': '0',
+    'aria-role': 'button',
+    '[attr.aria-controls]': 'parentPanel.getExpansionPanelId()',
+    '[attr.aria-expanded]': 'parentPanel.expanded',
+    '[class.mat-summary]': 'true',
+    '[class.expanded]': 'parentPanel.expanded',
+    '(click)': 'parentPanel.expanded = !parentPanel.expanded',
```

Comment:
It would be better if these expressions were in a method inside the class, instead of inline.